iOS/Objective C/IronSourceDemoApp/NativeAdView.m

```diff
@@ -0,0 +1,177 @@
+//
+//  NativeAdView.m
+//  IronSourceDemoApp
+//
+//  Created by Luga Wang on 9/15/23.
+//  Copyright © 2023 supersonic. All rights reserved.
+//
+
+#import "NativeAdView.h"
+
+@implementation NativeAdView
+
+@dynamic adAppIcon;
+@dynamic adTitleView;
+@dynamic adAdvertiserView;
+@dynamic adBodyView;
+@dynamic adMediaView;
+@dynamic adCallToActionView;
+
+- (instancetype)init {
+    UINib *nib = [UINib nibWithNibName:@"LevelPlayNativeAdView" bundle:[NSBundle mainBundle]];
+    NSArray *nibContents = [nib instantiateWithOwner:nil options:nil];
+    NativeAdView *nativeAdView = (NativeAdView *)[nibContents firstObject];
+    
+    if (!nativeAdView) {
+        nativeAdView = [[NativeAdView alloc] init];
+    }
+    
+    self = nativeAdView;
+    [self setNativeAdViewConstraints];
+    [self setupTopView];
+    self.translatesAutoresizingMaskIntoConstraints = NO;
+    
+    return self;
+}
+
+// This adding a customized top view to the NativeAdView that contains an adBadge on the left and a delete button on the right.
+// The delete button is to remove the native ad from the table view.
+- (void)setupTopView {
+    // Initialize UI elements
+    _adBadge = [[UILabel alloc] init];
+    _deleteButton = [UIButton buttonWithType:UIButtonTypeCustom];
+    _topView = [[UIView alloc] init];
+    
+    // Configure adBadge
+    _adBadge.text = @"AD";
+    _adBadge.font = [UIFont systemFontOfSize:10];
+    _adBadge.textAlignment = NSTextAlignmentCenter;
+    _adBadge.textColor = [UIColor whiteColor];
+    _adBadge.backgroundColor = [UIColor colorWithRed:0.0 green:0.0 blue:1.0 alpha:1.0]; // Replace with your desired color
+    _adBadge.layer.cornerRadius = 4.5;
+    _adBadge.clipsToBounds = YES;
+    _adBadge.translatesAutoresizingMaskIntoConstraints = NO;
+    
+    // Configure deleteButton
+    [_deleteButton setTitle:@"x" forState:UIControlStateNormal];
+    _deleteButton.backgroundColor = [UIColor lightGrayColor];
+    _deleteButton.layer.cornerRadius = 15;
+    [_deleteButton addTarget:self action:@selector(deleteButtonTapped:) forControlEvents:UIControlEventTouchUpInside];
+    _deleteButton.translatesAutoresizingMaskIntoConstraints = NO;
+    
+    // Add subviews
+    [_topView addSubview:_adBadge];
+    [_topView addSubview:_deleteButton];
+    _topView.translatesAutoresizingMaskIntoConstraints = NO;
+    [self addSubview:_topView];
+    
+    // Get the current top view
+    UIView *firstSubview = self.subviews[0];
+    
+    // Set up constraints
+    [NSLayoutConstraint activateConstraints:@[
```

Comment:
cant we add the constraint in the xib file?

Reply:
This one set up the topview constraints, now in (void) setTopViewConstraints(); should we keep it here since the top view is defined in class by code?

Reply:
yes